### PR TITLE
docs: how to customize event source

### PR DESCRIPTION
**Pull request by shaoshuai0102**
##### Checklist

- [ ] `npm test` passes
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message follows commit guidelines

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -51,13 +51,69 @@ Stop watching file(s).
...
+Choose to use your custom watching mode in your desired env.
+
+```js
+// config.${env}.js
+exports.watcher = {
+  type: 'custom',
+};
+```
+
+If possible, plugins named like `egg-watcher-${customName}`(`egg-watcher-vagrant` eg.) are welcomed.
```
Comment (atian25) [inline README.md]: - `plugins` -> 'plugin's'
- missing `eggPlugin.dependtion` description, these plugin will dep to `watcher`

**Comment by atian25** [comment]
do we need a guide doc about `watcher` at eggjs site?

File: README.md (hunk 1 above)
Comment (shaoshuai0102) [inline README.md]: - `plugins (named like..)  are welcomed.` You are not familiar with grammar, hah. :p
- watcher is built in. There's no need to claim deps

**Comment by shaoshuai0102** [comment]
It will add complexity of how developers think of egg if watcher appearing in the guide at this moment. If too much questions are asked about this, I will make a pr.

File: README.md (hunk 1 above)
Comment (atian25) [inline README.md]: yes, build-in plugin is always load before others, but declare it will be more clear.

anyway, +1

Comment (atian25) [inline README.md]: `not familiar with grammar` 😂

**Comment by atian25** [comment]
maybe just as a tutorial

File: README.md [hunk 2]
```diff
@@ -51,13 +51,69 @@ Stop watching file(s).
...
+  constructor(opts) {
+    super(opts);
+    this.ready(true);
+  }
+
+  watch(path) {
+    // replace this with your desired way of watching,
+    // when aware of any change, emit a `change` event
+    this._h = setInterval(() => {
+      this.emit('change', {
```
Comment (atian25) [inline README.md]: any other events or properties to describe `add`/`remove` etc.

Comment (shaoshuai0102) [inline README.md]: We reduced events to just `change` to simplify watcher plugin development. In common scenarios, once `change` triggered, just clearing all view template cache is the best choice. It's simple and robust in most cases.

Comment (atian25) [inline README.md]: yes, but if offer a `type` to `info` object will be nice to developer.

what I mean is should we define the suggest properties of `info` object, so when developer write their custom watcher, can follow the same rule.

Comment (shaoshuai0102) [inline README.md]: Added

**Pull request merged**